src/java/org/apache/cassandra/db/Directories.java

```diff
@@ -685,6 +719,28 @@ public static File getBackupsDirectory(File location)
         }
     }
 
+    /**
+     * Returns directory to write a backup to. If directory does not exist, then it is NOT created.
+     *
+     * If given {@code location} indicates secondary index, this will return
+     * {@code <cf dir>/backups/.<index name>}.
+     * Otherwise, this will return {@code <cf dir>/backups/}.
+     *
+     * @param location base directory
+     * @return directory to write a backup
+     */
+    public static Optional<File> getBackupsDirectoryIfExists(File location)
```

Comment:
I know it is probably unrelated or too late but I don't really like the fact that the table's directory structure leaks outside the `Descriptor` class.